homeassistant/components/cloud/account_link.py

```diff
@@ -64,8 +64,7 @@ async def _get_services(hass: HomeAssistant) -> list[dict[str, Any]]:
     """Get the available services."""
     services: list[dict[str, Any]]
     if DATA_SERVICES in hass.data:
-        services = hass.data[DATA_SERVICES]
-        return services
+        return cast(list[dict[str, Any]], hass.data[DATA_SERVICES])
```

Comment:
Here we have the option to either use a cast or to add a noqa.

Reply:
But I think I will add an issue to Ruff, so that this may not be claimed in the future.

Reply:
Yes, when the typing is at the assignment directly before the return, it doesn't get flagged, but this case here where the typing is further above is getting flagged.

Reply:
Thanks to https://github.com/astral-sh/ruff/pull/10741 altering the behavior, I changed it now to a `noqa` until a Ruff version releases with the change incorporated, after which the `noqa` can be removed.